pkg/api/v1alpha1/vspheremachineconfig_webhook.go

```diff
@@ -91,6 +84,19 @@ func validateImmutableFieldsVSphereMachineConfig(new, old *VSphereMachineConfig)
 		)
 	}
 
+	if !old.IsControlPlane() && !old.IsEtcd() {
+		vspheremachineconfiglog.Info("Machine config is not associated with control plane or etcd")
+		return allErrs
+	}
+	vspheremachineconfiglog.Info("Machine config is associated with control plane or etcd")
+
+	if old.Spec.Template != new.Spec.Template {
```

Comment:
We probably should not make this mutable until we are able to auto import templates and ensure that this won't break worker node configuration?